core/dialog.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview Wrapper functions around JS functions for showing
+ * alert/confirmation dialogs.
+ */
+
+'use strict';
+
+goog.module('Blockly.dialog');
+goog.module.declareLegacyNamespace();
+
+let alertImplementation = function(message, opt_callback) {
+  window.alert(message);
+  if (opt_callback) {
+    opt_callback();
+  }
+};
+
+let confirmImplementation = function(message, callback) {
+  callback(window.confirm(message));
+};
+
+let promptImplementation = function(message, defaultValue, callback) {
+  callback(window.prompt(message, defaultValue));
+};
+
+/**
+ * Wrapper to window.alert() that app developers may override to
+ * provide alternatives to the modal browser window.
+ * @param {string} message The message to display to the user.
+ * @param {function()=} opt_callback The callback when the alert is dismissed.
+ */
+const alert = function(message, opt_callback) {
```

Comment:
Why create a wrapper here, instead of just doing `exports.alert = alertImplementation`?

Reply:
Because that exports the initial value of alertImplementation, so even if alertImplementation gets modified via the setter, the initial/default implementation will continue to be used.

Reply:
Right, that makes sense.

The comment is no longer correct (for this and the others) because to override you would call `setAlert` rather than overriding `alert`.

Reply:
Updated the JSDoc to clarify, thanks!